### PR TITLE
T1 2414 multiseason

### PR DESCRIPTION
**Pull request by pietro-diana**

### CODE REVIEW

File: season/Seasons.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    return Promise.resolve(this)
+  }
+
+  getAllSeasons() {
+    return this.seasons
+  }
+
+  getFirstSeason() {
+    if (this.seasons.length) return this.seasons[0]
+    return null
```
Comment (fewhnhouse) [inline season/Seasons.js:52]: return this.seasons.length > 0 ? this.seasons[0] : null

File: season/Seasons.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  getSeasonById(seasonId) {
+    return this.seasons.find((element) => element.id === seasonId)
```
Comment (fewhnhouse) [inline season/Seasons.js:56]: `season => season.id`

File: season/Season.js [hunk 3]
```diff
@@ -1,43 +1,24 @@
...
-
-  parseProductSeasons(rawProductSeasons) {
-    for (let rawProductSeason of rawProductSeasons) {
-      this.productSeasons.push(new ProductSeason(rawProductSeason))
-    }
+  constructor(params) {
+    this.id = params && params.id ? params.id : null
+    this.from = params && params.from ? new Date(params.from) : null
+    this.to = params && params.to ? new Date(params.to) : null
+    this.name = params && params.name ? params.name : null
```
Comment (fewhnhouse) [inline season/Season.js:6]: `params.name || null`
For the others too.

File: season/Seasons.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import Season from '@/classes-shared/season/Season'
+import { peInstance } from '../utils/axiosInstance'
+
+export default class Seasons {
+  constructor(param = {}) {
+    this.seasons = param.seasons ? param.seasons : []
```
Comment (fewhnhouse) [inline season/Seasons.js:6]: `param.seasons || []`

**Review by fewhnhouse — CHANGES_REQUESTED** [review]

File: season/Season.js (hunk 3 above)
Comment (tobias-erpen) [reply season/Season.js:6]: Only for id and name right. or how would it work with the new Date()?

File: season/Seasons.js (hunk 2 above)
Comment (tobias-erpen) [reply season/Seasons.js:56]:
```
  getSeasonById(season) {
    return this.seasons.find((element) => element.id === season.id)
  }
```
like that?

File: season/Season.js (hunk 3 above)
Comment (fewhnhouse) [reply season/Season.js:6]: Correct.

File: season/Seasons.js (hunk 2 above)
Comment (tobias-erpen) [reply season/Seasons.js:56]: ahh now I get it

Comment (fewhnhouse) [reply season/Seasons.js:56]: No, like this:
`this.seasons.find((season) => season.id === seasonId)`

The name `element` is not explicit.

**Review by fewhnhouse — APPROVED** [review]

**Pull request merged**